docs/TELEMETRY.md

```diff
@@ -0,0 +1,131 @@
+
+Some telemetry services may have limited number of dimensions. For example, Google Analytics has a limit of 20 custom dimensions. As Web Chat may introduce more dimensions later, developers are advised to pick dimensions they needed before sending to their services.
+
+### Events for hooks
+
+When the following hooks are called, one or more event measurements will be emitted.
+
+| Hooks              | Events                  | Description                                               |
+| ------------------ | ----------------------- | --------------------------------------------------------- |
+| `useSendFiles`     | `sendFiles`             | Emit when the user uploading files                        |
+| `useSendFiles`     | `sendFiles:numFile`     | Number of files uploaded                                  |
+| `useSendFiles`     | `sendFiles:sumSizeInKB` | Total file size in kilobytes                              |
+| `useSubmitSendBox` | `submitSendBox`         | `data` is the method of submit, for example, `"keyboard"` |
+
+### Exceptions
+
+### Timings
+
+The following operations are timed:
+
+| Timing                    | Description                                                                       |
+| ------------------------- | --------------------------------------------------------------------------------- |
+| `sendFiles:makeThumbnail` | Time used to generate thumbnail for every uploading image via `useSendFiles` hook |
+
+## Data collection
+
+To collect measurements, you will need to pass an `onTelemetry` handler as a prop to Web Chat. The event emitted will be one of the types below:
+
+```ts
+interface TelemetryMeasurementEvent {
+   type: string;
+   dimension: any;
+}
+
+interface TelemetryEventMeasurementEvent extends TelemetryMeasurementEvent {
+   type: 'event';
+   name: string;
+   data?: string;
+   value?: number;
+}
+
+interface TelemetryExceptionMeasurementEvent extends TelemetryMeasurementEvent {
+   type: 'exception';
+   error: Error;
+}
+
+interface TelemetryTimingStartMeasurementEvent extends TelemetryMeasurementEvent {
+   type: 'timingstart';
+   name: string;
+   timingId: string;
+}
+
+interface TelemetryTimingEndMeasurementEvent extends TelemetryMeasurementEvent {
+   type: 'timingend';
+   name: string;
+   timingId: string;
+   duration: number;
+}
+```
+
+## Hooks
+
+To emit custom measurements through the `onTelemetry` handler, you can use one of the following hooks.
+
+### Events
+
+```ts
+useTrackEvent() =>
+  (
+    name: string,
+    data?: string
+  ) => void;
+```
+
+### Exceptions
+
+```ts
+useTrackException() =>
+  (
+    error: Exception
+  ) => void;
+```
+
+### Timing
+
+```ts
+useTrackTiming() =>
+  (
+    name: string
+  ) =>
+    (() => void | Promise);
+```
+
+## Samples
+
+### Google Analytics
+
+-  20 custom dimensions
+-  500 event hits per session
```

Comment:
Just in case Google changes their specs, I would recommend simply linking to their info page here :)